src/apis/doctor/doctor.service.js

```diff
@@ -186,7 +177,7 @@ class DoctorService {
             condition = condition + `dst.value = '${location}'`;
         }
         if (speciality) {
-            condition = condition + ` AND d.speciality = '${speciality}'`;
+            condition = condition + ` AND d.speciality LIKE CONCAT('%','${speciality}','%')`;
```

Comment:
What is the need of this?

Reply:
because we  cannot do an  exact match as speciality is and JSON object we should use pattern matching to sort exact match

Reply:
Specialty will be matched exactly because the field is provided by our end from the master data table. There is no possibility for other specialty that could be stored or in other words there can not be another string value that could be passed.

Reply:
is that not our expected behaviour? what should  i change here?

Reply:
There was no need to change the query here. You can revert it back